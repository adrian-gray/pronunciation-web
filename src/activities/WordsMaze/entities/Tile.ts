const DISABLED_TINT = 0.5;

const DEFAULT = 0xdddddd;
const STROKE = 0x000000;
const INCORRECT = 0xffcccc;
const CORRECT = 0xccffcc;
const YELLOW = 0xffffcc;
const BLUE = 0xccccff;
const STACK = "'Helvetica Neue', Helvetica, Arial, sans-serif";

interface TileParams {
  index: number;
  state: Phaser.State;
  $tiles: Phaser.Signal;
  parent: Phaser.Group;
  TILE_SIZE: number;
  row: number;
  col: number;
  word: string;
}

export default function Tile(params: TileParams) {
  const { index, state, $tiles, parent, TILE_SIZE, row, col, word } = params;

  const group = state.add.group();
  parent.add(group);
  const back = makeBack();
  back.events.onInputUp.add(handleUp);
  back.inputEnabled = true;

  const text = addText({ str: word, size: 48 });

  group.add(back);
  group.add(text);

  function makeBack() {
    const g = state.add.graphics(col * TILE_SIZE, row * TILE_SIZE);
    let color = DEFAULT;
    if (index === 0) color = YELLOW;
    if (index === 71) color = BLUE;

    // set a fill and line style
    g.beginFill(color);
    g.lineStyle(2, STROKE, 1);

    g.drawRect(0, 0, TILE_SIZE, TILE_SIZE);
    return g;
  }

  function addText({ str, size }: { str: string; size: number }) {
    const style = { font: `normal ${size}px ${STACK}` };
    const posX = col * TILE_SIZE + TILE_SIZE / 2;
    const posY = row * TILE_SIZE + TILE_SIZE / 2;
    const text = state.add.text(posX, posY, str, style);
    text.anchor.set(0.5);
    return text;
  }

  function handleUp() {
    $tiles.dispatch({ word, index, row, col });
  }

  function correct() {
    back.tint = CORRECT;
    disable();
  }

  function incorrect() {
    back.tint = INCORRECT;
    disable();
  }

  function disable() {
    text.alpha = DISABLED_TINT;
    back.inputEnabled = false;
  }

  const API = {
    correct,
    incorrect
  };

  return API;
}
