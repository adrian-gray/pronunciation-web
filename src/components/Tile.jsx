import React, { Component } from 'react'

import {
  Paper,
  Typography,
  withStyles
} from '@material-ui/core'

const styles = (theme) => ({
  largeText: theme.largeText,
  left: {
    ...theme.subCanvas,
    float: 'left',
    margin: '2rem',
    padding: '1rem'
  },
  clickable: {
    cursor: 'pointer'
  },
  correct: {
    backgroundColor: '#CCFFCC'
  },
  incorrect: {
    backgroundColor: '#FFCCCC'
  }
})

class Tile extends Component {
  constructor (props) {
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (e) {
    this.props.handleClick(this.props.index)
  }

  render () {
    const { classes } = this.props
    const paperClasses = [classes.left]

    let correctStatus = null
    if (this.props.isCorrect) {
      paperClasses.push(classes.correct)
    } else if (this.props.isCorrect === false) {
      paperClasses.push(classes.incorrect)
    }

    return (
      <div className={classes.clickable} onClick={this.handleClick}>
        <Paper className={paperClasses.join(' ')}>
          <Typography className={classes.largeText}>
            <span className={classes[correctStatus]}>
              {this.props.word}
            </span>
          </Typography>
        </Paper>
      </div>
    )
  }
}

export default withStyles(styles)(Tile)
