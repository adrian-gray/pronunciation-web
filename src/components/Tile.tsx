import React from "react";

export default props => {
  const handleClick = e => {
    props.handleClick(props.index);
  };

  const paperClasses = ["tile"];

  let correctStatus = null;
  if (props.isCorrect) {
    paperClasses.push("correct-bg");
    correctStatus = "correct-bg";
  } else if (props.isCorrect === false) {
    paperClasses.push("incorrect-bg");
    correctStatus = "incorrect-bg";
  }

  return (
    <div className="cursor-pointer" onClick={handleClick}>
      <div className={paperClasses.join(" ")}>
        <p className="large-text">
          <span className={correctStatus}>{props.word}</span>
        </p>
      </div>
    </div>
  );
};
