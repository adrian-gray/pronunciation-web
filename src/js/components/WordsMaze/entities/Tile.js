const DISABLED_TINT = 0.5

const DEFAULT = 0xDDDDDD
const STROKE = 0x000000
const INCORRECT = 0xFFCCCC
const CORRECT = 0xCCFFCC
const STACK = "'Helvetica Neue', Helvetica, Arial, sans-serif"

export default function Tile (params) {
  const { index, state, $tiles, parent, pos, word } = params
  const { x, y, width, height } = pos

  const group = state.add.group()
  parent.add(group)
  const back = makeBack()
  back.events.onInputUp.add(handleUp)
  back.inputEnabled = true

  const text = addText({ str: word, size: 54 })

  group.addMultiple(back, text)

  function makeBack () {
    const g = state.add.graphics(x, y)

    // set a fill and line style
    g.beginFill(DEFAULT)
    g.lineStyle(2, STROKE, 1)

    g.drawRect(0, 0, width, height, 5)
    return g
  }

  function addText ({ str, size }) {
    const style = { font: `normal ${size}px ${STACK}` }
    const posX = x + (width / 2)
    const posY = y + (height / 2)
    const text = state.add.text(posX, posY, str, style)
    text.anchor.set(0.5)
    return text
  }

  function handleUp () {
    $tiles.dispatch({ word, index })
  }

  function correct () {
    back.tint = CORRECT
    disable()
  }

  function incorrect () {
    back.tint = INCORRECT
    disable()
  }

  function disable () {
    text.alpha = DISABLED_TINT
    back.inputEnabled = false
  }

  const API = {
    correct,
    incorrect
  }

  return API
}
