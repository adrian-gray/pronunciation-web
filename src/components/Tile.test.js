import React from "react";
import { render, cleanup } from "@testing-library/react";

import Tile from "./Tile";

afterEach(cleanup);

test("Tile", async () => {
  const props = {
    handleClick: () => "word",
    isCorrect: true,
    index: 1,
    word: "Grease"
  };

  const { container } = render(<Tile {...props} />);

  expect(container).toMatchSnapshot();
});
