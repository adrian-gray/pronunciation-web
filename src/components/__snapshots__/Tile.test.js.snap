// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile 1`] = `
<div>
  <div
    class="makeStyles-clickable-5"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-left-4 makeStyles-correctBG-2 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root makeStyles-largeText-1 MuiTypography-body1"
      >
        <span
          class="makeStyles-correctBG-2"
        >
          Grease
        </span>
      </p>
    </div>
  </div>
</div>
`;
